packages/dashboard/src/components/CompositeList/CompositeList.tsx

```diff
@@ -0,0 +1,230 @@
+          shimmerLines={20}
+          enableShimmer={this.props.detailsListProps.items.length === 0}
+          componentRef={this.detailsListRef}
+          layoutMode={DetailsListLayoutMode.fixedColumns}
+          constrainMode={ConstrainMode.unconstrained}
+          onShouldVirtualize={this.defaultOnShouldVirtualize}
+          useReducedRowRenderer={useReducedRowRenderer}
+          onRenderRow={this.defaultOnRenderRow}
+          className={mergeStyles(getDetailsListStyle(!!this.props.overrideStickyPosition))}
+          {...this.props.detailsListProps}
+          styles={this.props.detailsListProps.styles}
+          selection={this.selection}
+          compact={this.isCompactMode}
+          columns={wrappedColumns}
+        />
+      </>
+    );
+  }
+
+  public componentDidMount(): void {
+    stickyfill.addOne(this.commandBarRef.current!);
+
+    // tslint:disable-next-line:no-any
+    const titleWrapper = (ReactDOM.findDOMNode((this.detailsListRef.current! as any) as React.ReactInstance) as Element).querySelector(
+      '.ms-DetailsList-headerWrapper'
+    );
+
+    if (titleWrapper) {
+      stickyfill.addOne(titleWrapper as HTMLElement);
+    }
+  }
+
+  private readonly defaultOnRenderRow = (props: IDetailsRowProps, defaultRender: IRenderFunction<IDetailsRowProps>): JSX.Element => {
+    return <div>{defaultRender({ ...props })}</div>;
+  };
+
+  private get isCompactMode(): boolean {
+    return this.props.isCompactMode || this.state.isCompactMode;
+  }
+  private get numberOfItemsSelected(): number {
+    return this.props.numberOfItemsSelected || this.state.numberOfItemsSelected;
+  }
+  private readonly defaultOnShouldVirtualize = () => this.props.detailsListProps.items.length >= 50;
+  private readonly defaultOnClickSelected = () => this.selection.setAllSelected(false);
+  private readonly defaultNormalListOnClick = () => {
+    this.setState({
+      isCompactMode: false
+    });
+  };
+  private readonly defaultCompactListOnClick = () => {
+    this.setState({
+      isCompactMode: true
+    });
+  };
+  private readonly defaultFarItemsFactory = (options: ICompositeListFarItemOptions) => {
+    const getSelectedText = options.getSelectedText || ((count: number) => `${count} selected`);
+    const getSelectedAriaLable = options.getSelectedAriaLabel || ((count: number) => `${count} selected items`);
+
+    const mergedCommandBarItemStyle = mergeStyleSets(
+      getCommandBarItemStyle(),
+      options.compactMenuProps ? options.compactMenuProps.buttonStyles : undefined
+    );
+
+    const farItems: ICommandBarItemProps[] = [
+      {
+        key: 'search',
+        onRender: (): React.ReactNode => {
+          const mergedCommandBarSearchBoxStyle = mergeStyleSets(
+            getCommandBarSearchBoxStyle(),
+            options.searchBoxProps ? options.searchBoxProps.styles : undefined
+          );
+          return (
+            <SearchBox
+              ariaLabel={'Input for searching this list'}
```

Comment:
>{'Input for searching this list'} [](start = 24, length = 33)

take this as props, as this will not be localized in fabric

Reply:
no. Intentional like this, user can override it.